producer/src/main/kotlin/streams/kafka/KafkaEventRouter.kt

```diff
@@ -42,43 +43,44 @@ class KafkaEventRouter: StreamsEventRouter {
         producer.close()
     }
 
-    private fun send(producerRecord: ProducerRecord<String, ByteArray>) {
-        producer.send(producerRecord) { meta: RecordMetadata?, error: Exception? ->
+    private fun send(producerRecord: ProducerRecord<String, ByteArray>): Future<RecordMetadata> {
+        return producer.send(producerRecord) { meta: RecordMetadata?, error: Exception? ->
             if (log.isDebugEnabled) {
                 log.debug("Sent record in partition ${meta?.partition()} offset ${meta?.offset()} data ${meta?.topic()} key size ${meta?.serializedKeySize()}", error)
             }
         }
     }
 
-    private fun sendEvent(partition: Int, topic: String, event: StreamsEvent) {
+    private fun sendEvent(partition: Int, topic: String, event: StreamsEvent): Future<RecordMetadata> {
         if (log.isDebugEnabled) {
             log.debug("Trying to send a simple event with payload ${event.payload} to kafka")
         }
         val uuid = UUID.randomUUID().toString()
         val producerRecord = ProducerRecord(topic, partition, System.currentTimeMillis(), uuid,
                 JSONUtils.writeValueAsBytes(event))
-        send(producerRecord)
+        return send(producerRecord)
     }
 
-    private fun sendEvent(partition: Int, topic: String, event: StreamsTransactionEvent) {
+    private fun sendEvent(partition: Int, topic: String, event: StreamsTransactionEvent): Future<RecordMetadata> {
         if (log.isDebugEnabled) {
             log.debug("Trying to send a transaction event with txId ${event.meta.txId} and txEventId ${event.meta.txEventId} to kafka")
         }
         val producerRecord = ProducerRecord(topic, partition, System.currentTimeMillis(), "${event.meta.txId + event.meta.txEventId}-${event.meta.txEventId}",
                 JSONUtils.writeValueAsBytes(event))
-        send(producerRecord)
+        return send(producerRecord)
     }
 
-    override fun sendEvents(topic: String, transactionEvents: List<out StreamsEvent>) {
-        try {
+    override fun sendEvents(topic: String, transactionEvents: List<out StreamsEvent>): List<out Future<RecordMetadata>> {
```

Comment:
Instead using futures I would want to keep this `async` and then use `use await sendEvents()` in the sync method.
@conker84 will look into this.